content/blog/2014/sqlalchemy-postgres-autocommit.md

```diff
@@ -226,6 +259,66 @@ the database. But we don't actually need to issue `BEGIN` ourselves
 either - we just need to turn off the `autocommit` property on our
 connection, and then `psycopg2` will issue the `BEGIN` for us.
 
+{% callout 'note', update %}
+SQLAlchemy 2.0 now [recommends] starting an "autocommit" connection when needed,
+and using the regular connections otherwise:
+
+```python
+# autocommit block, no need to begin() or commit()
+with engine.connect().execution_options(isolation_level="AUTOCOMMIT") as connection:
+    connection.execute(text("<statement>"))
+
+# regular block, needs begin() and calls commit() at exit
+with engine.begin() as connection:
+    connection.execute(text("<statement>"))
+```
```

Comment:
@mirisuzanne Do you think we should add different styles for these code blocks inside callouts? Just below this: https://deploy-preview-577--oddleventy.netlify.app/2014/06/14/sqlalchemy-postgres-autocommit/#turning-on-real-autocommit

Reply:
Yeah, we should probably just remove the negative margins (or avoid adding them when we're in a callout if we want to exclude instead of overriding)